src/main/scala/com/microsoft/hyperspace/index/IndexLogEntry.scala

```diff
@@ -56,6 +56,16 @@ case class Content(root: Directory, fingerprint: NoOpFingerprint = NoOpFingerpri
         FileInfo(new Path(prefix, f.name).toString, f.size, f.modifiedTime, f.id)).toSet
   }
 
+  @JsonIgnore
+  lazy val versionInfos: Set[Int] = {
```

Comment:
In the log entry, it is hard to find which versions are actually used. 
Maybe it is better to track ids that are currently used in a class variable of IndexLogEntry (or maybe LogEntry).

Reply:
Could you utilize `private def getIndexContentDirectoryPaths(entry: IndexLogEntry): Seq[String] = {` in IndexStatistics?
It would be good to move the function to here.